Resolves jax bug:

```
TensorFlow.js: export failure ❌ 0.2s: cannot import name 'shape_poly' from 'jax.experimental.jax2tf' (/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/jax/experimental/jax2tf/__init__.py)
Traceback (most recent call last):
  File "/home/runner/work/ultralytics/ultralytics/ultralytics/utils/benchmarks.py", line 105, in benchmark
    filename = model.export(imgsz=imgsz, format=format, half=half, int8=int8, device=device, verbose=False)
  File "/home/runner/work/ultralytics/ultralytics/ultralytics/engine/model.py", line 310, in export
    return Exporter(overrides=args, _callbacks=self.callbacks)(model=self.model)
  File "/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/home/runner/work/ultralytics/ultralytics/ultralytics/engine/exporter.py", line 271, in __call__
    f[9], _ = self.export_tfjs()
  File "/home/runner/work/ultralytics/ultralytics/ultralytics/engine/exporter.py", line 122, in outer_func
    raise e
  File "/home/runner/work/ultralytics/ultralytics/ultralytics/engine/exporter.py", line 117, in outer_func
    f, model = inner_func(*args, **kwargs)
  File "/home/runner/work/ultralytics/ultralytics/ultralytics/engine/exporter.py", line 786, in export_tfjs
    import tensorflowjs as tfjs  # noqa
  File "/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/tensorflowjs/__init__.py", line 21, in <module>
    from tensorflowjs import converters
  File "/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/tensorflowjs/converters/__init__.py", line 26, in <module>
    from tensorflowjs.converters.jax_conversion import convert_jax
  File "/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/tensorflowjs/converters/jax_conversion.py", line 20, in <module>
    from jax.experimental.jax2tf import shape_poly
ImportError: cannot import name 'shape_poly' from 'jax.experimental.jax2tf' (/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/jax/experimental/jax2tf/__init__.py)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/runner/work/ultralytics/ultralytics/ultralytics/cfg/__init__.py", line 465, in <module>
    entrypoint(debug='')
  File "/home/runner/work/ultralytics/ultralytics/ultralytics/cfg/__init__.py", line [448](https://github.com/ultralytics/ultralytics/actions/runs/7208624405/job/19637952839#step:6:449), in entrypoint
    getattr(model, mode)(**overrides)  # default args from model
  File "/home/runner/work/ultralytics/ultralytics/ultralytics/engine/model.py", line 289, in benchmark
    return benchmark(
  File "/home/runner/work/ultralytics/ultralytics/ultralytics/utils/benchmarks.py", line 131, in benchmark
    assert type(e) is AssertionError, f'Benchmark failure for {name}: {e}'
AssertionError: Benchmark failure for TensorFlow.js: cannot import name 'shape_poly' from 'jax.experimental.jax2tf' (/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/jax/experimental/jax2tf/__init__.py)
```

--- a/requirements.txt
+++ b/requirements.txt
@@ -32,8 +32,8 @@ seaborn>=0.11.0
 # scikit-learn==0.19.2  # CoreML quantization
 # tensorflow>=2.4.1,<=2.13.1  # TF exports (-cpu, -aarch64, -macos)
 # tflite-support
-# jax <= 0.4.21   # for tensorflowjs bug https://github.com/google/jax/issues/18978
-# jaxlib <= 0.4.21   # for tensorflowjs bug https://github.com/google/jax/issues/18978
+# jax <= 0.4.21   # tensorflowjs bug https://github.com/google/jax/issues/18978
+# jaxlib <= 0.4.21   # tensorflowjs bug https://github.com/google/jax/issues/18978
 # tensorflowjs>=3.9.0  # TF.js export
 # openvino-dev>=2023.0  # OpenVINO export
 

--- a/requirements.txt
+++ b/requirements.txt
@@ -32,8 +32,8 @@ seaborn>=0.11.0
 # scikit-learn==0.19.2  # CoreML quantization
 # tensorflow>=2.4.1,<=2.13.1  # TF exports (-cpu, -aarch64, -macos)
 # tflite-support
-# jax <= 0.4.21   # for tensorflowjs
-# jaxlib <= 0.4.21   # for tensorflowjs
+# jax <= 0.4.21   # for tensorflowjs bug https://github.com/google/jax/issues/18978
+# jaxlib <= 0.4.21   # for tensorflowjs bug https://github.com/google/jax/issues/18978
 # tensorflowjs>=3.9.0  # TF.js export
 # openvino-dev>=2023.0  # OpenVINO export
 

--- a/ultralytics/engine/exporter.py
+++ b/ultralytics/engine/exporter.py
@@ -781,7 +781,8 @@ class Exporter:
     @try_export
     def export_tfjs(self, prefix=colorstr('TensorFlow.js:')):
         """YOLOv8 TensorFlow.js export."""
-        check_requirements('tensorflowjs')
+        # JAX bug requiring install constraints in https://github.com/google/jax/issues/18978
+        check_requirements(['jax<=0.4.21', 'jaxlib<=0.4.21', 'tensorflowjs'])
         import tensorflow as tf
         import tensorflowjs as tfjs  # noqa
 

